SimPEG/simulation.py

```diff
@@ -524,40 +785,98 @@ def G(self):
 
     @G.setter
     def G(self, G):
-        # Allows setting G in a LinearSimulation
+        # Allows setting G in a LinearSimulation.
         # TODO should be validated
         self._G = G
 
     def fields(self, m):
+        # Docstring inherited from BaseSimulation.
         self.model = m
         return self.G.dot(self.linear_model)
 
     def dpred(self, m=None, f=None):
+        # Docstring inherited from BaseSimulation
         if m is not None:
             self.model = m
         if f is not None:
             return f
         return self.fields(self.model)
 
     def getJ(self, m, f=None):
+        r"""Returns the full Jacobian.
+
+        The general definition of the linear forward simulation is:
+
+        .. math::
+            \mathbf{d} = \mathbf{G \, f}(\mathbf{m})
+
+        where :math:`\mathbf{f}` is a mapping operator (optional) from the model space
+        to a user-defined parameter space, and :math:`\mathbf{G}` is an (n_data, n_param)
+        linear operator. The ``getJ`` method forms and returns the full Jacobian:
+
+        .. math::
+            \mathbf{J}(\mathbf{m}) = \mathbf{G} \frac{\partial \mathbf{f}}{\partial \mathbf{m}}
+
+        for the model :math:`\mathbf{m}` provided. When :math:`\mathbf{f}` is the identity map
+        (default), the Jacobian is no longer model-dependent and reduces to:
+
+        .. math::
+            \mathbf{J} = \mathbf{G}
+
+        Parameters
+        ----------
+        m : numpy.ndarray
+            The model vector.
+        f : None
+            Precomputed fields are not used to speed up the computation of the
+            Jacobian for linear problems.
+
+        Returns
+        -------
+        J : (n_data, n_param) numpy.ndarray
+            :math:`J = G\frac{\partial f}{\partial\mathbf{m}}`.
+            Where :math:`f` is :attr:`model_map`.
+        """
         self.model = m
         # self.model_deriv is likely a sparse matrix
         # and G is possibly dense, thus we need to do..
         return (self.model_deriv.T.dot(self.G.T)).T
 
     def Jvec(self, m, v, f=None):
+        # Docstring inherited from BaseSimulation
         self.model = m
         return self.G.dot(self.model_deriv * v)
 
     def Jtvec(self, m, v, f=None):
+        # Docstring inherited from BaseSimulation
         self.model = m
         return self.model_deriv.T * self.G.T.dot(v)
 
 
 class ExponentialSinusoidSimulation(LinearSimulation):
```

Comment:
While updating this branch with the latest changes in `main` I had to resolve a few issues in this file. One of them was merging the changes that @ghwilliams and @lheagy made to the docstring of the `ExponentialSinusoidSimulation` class (@lheagy applied those changes in #1337).

I kept the two additions to avoid deleting anything important. But maybe there is room for improvement and better merging the two ideas. Would you take a second look at this?

Reply:
For sure. I'm looking forward to seeing how the new doc will look. Thanks.